packages/vscode-extension/src/webview/components/UrlSelect.tsx

```diff
@@ -35,7 +36,7 @@ function UrlSelect({ onValueChange, recentItems, items, value, disabled }: UrlSe
   };
 
   return (
-    <Select.Root onValueChange={handleValueChange} value={value} disabled={disabled}>
+    <Select.Root onValueChange={handleValueChange} key={key} value={value} disabled={disabled}>
```

Comment:
It should be enough to do something along these lines, instead of having the whole logic for urlSelectKey in the parent component:

```tsx
<Select.root key={value ? "has-value : "no-value"}>
```